**Q1) A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.**

     Minitab File : Cutlets.mtw


In [4]:
#Importing Libraries 
import pandas as pd
import scipy
from scipy import stats

In [5]:
#Load dataset
cutlet = pd.read_csv('/content/Cutlets.csv')
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [6]:
#Descriptive statistics
cutlet.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [7]:
#Changing column name
cutlet.columns = 'unit_a','unit_b'

In [8]:
#P-values : 2-sample, 2-tail ttest : stats.ttest_int(array1,array2)       #ind -> Independant Variable
p_value = stats.ttest_ind(cutlet.unit_a,cutlet.unit_b)
p_value


Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

**Q2) A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.**
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 
 
    Minitab File: LabTAT.mtw


In [9]:
        import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

#Load the dataset
labtat = pd.read_csv('/content/LabTAT.csv')
labtat

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [10]:
# Anova ftest 
x = 0.05            #5% significance level
p_value = stats.f_oneway(labtat.iloc[:,0],labtat.iloc[:,1],labtat.iloc[:,2],labtat.iloc[:,3])
print(p_value)

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)


p_value (2.1156) > 0.05               #5% significance level

so Choose null hypothesis and reject alternative          

**Q3) Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.**

In [11]:
#Import libraries
import pandas as pd
from scipy import stats
from scipy.stats import norm

#Load the data set
df = pd.read_csv('/content/BuyerRatio.csv')
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [12]:
df_table = df.iloc[:,1:6]
df_table

East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [13]:
df_table.values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [14]:
val = stats.chi2_contingency(df_table)
val

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [15]:
degree_of_fd = ((len(df_table.iloc[0:2,0]))-1) * ((len(df_table.iloc[0,0:4]))-1)
print('Degree_of_freedom :',degree_of_fd)

Degree_of_freedom : 3


In [16]:
Expected_value = val[3]
Expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [17]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2/e for o,e in zip(df_table.values,Expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [18]:
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

In [19]:
if chi_square_statestic >= critical_value:
  print('Dependent (reject null hypothesis (H0))')
else:
  print('Independent (failed to reject null hypothesis (H0))')


Independent (failed to reject null hypothesis (H0))


In [20]:
pvalue = 1-chi2.cdf(chi_square_statestic,3)
print(pvalue)

0.6787446296467897


In [21]:
if pvalue <= 0.05:
   print('Dependent (reject null hypothesis (H0))')
else:
  print('Independent (failed to reject null hypothesis (H0))')


Independent (failed to reject null hypothesis (H0))


**Q4) TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences**

# Minitab File: CustomerOrderForm.mtw 

Assume Null Hypothesis as Ho: Independence of categorical variables
 
Alternative hypothesis as Ha Dependence of categorical variables 

In [22]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [23]:
#load the dataset
data = pd.read_csv('/content/Costomer+OrderForm.csv')
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [24]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [25]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [26]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [27]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [28]:
obs = np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [29]:
chi2_contingency(obs)      #output is chi2_stat_value, p_value, degree_of_freedom and expected observations

Chi2ContingencyResult(statistic=3.858960685820355, pvalue=0.2771020991233135, dof=3, expected_freq=array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]]))

Inference: As (p_value = 0.2771) > (alpha = 0.05); 

Accept Null Hypthesis 

i.e.Independence of categorical variables Thus, customer order forms defective % does not varies by centre